src/utils/opt_in_checker.py

```diff
@@ -0,0 +1,260 @@
+from .input_with_timeout import input_with_timeout
+
+
+class ISIPCheckResult(Enum):
+    DECLINED = 0
+    ACCEPTED = 1
+    NO_FILE = 2
+
+
+class DialogResult(Enum):
+    DECLINED = 0
+    ACCEPTED = 1
+    TIMEOUT_REACHED = 2
+
+
+class OptInChecker:
+    dialog_timeout = 50  # seconds
+    path_to_opt_in_out_script = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+    opt_in_out_script_name = "opt_in_out"
+    doc_link = "docs.openvino.ai"
+    opt_in_out_script_run_command = "\'{} --opt_out\'".format(opt_in_out_script_name)
+    opt_in_question = "With your permission, Intel would like to collect occasional (anonymous) tool usage such " \
+                      "as software installation, development environment, and usage responses to improve our product " \
+                      "and your developer experience.  This confidential data will be kept a period of time " \
+                      "to study trends, and of course you can opt out at any time in the future by running {}. " \
+                      "We really appreciate your help by accepting (just type \"Y\" at the prompt) to show your " \
+                      "consent to the collection of this data, a \"N\" will prevent the anonymous data from " \
+                      "being sent back to Intel. " \
+                      "Thanks for replying with a \"Y\"!".format(opt_in_out_script_run_command)
+    opt_in_question = "Intel would like your permission to collect software performance and usage data for the " \
+                      "purpose of improving Intel products and services. This data will be collected directly " \
+                      "by Intel or through the use of Google Analytics. This data will be stored in countries " \
+                      "where Intel or Google operate.\nYou can opt-out at any time in the future by running {}.\n" \
+                      "More Information is available at {}.\n" \
+                      "Please type ‘Y’ to give your consent or ‘N’ to decline.".format(opt_in_out_script_run_command,
+                                                                                       doc_link)
+    opt_in_question_incorrect_input = "Please type ‘Y’ to give your consent or ‘N’ to decline."
+    response_confirmation_accept = "The selected option was to collect telemetry data."
+    response_confirmation_decline = "The selected option was NOT to collect telemetry data."
+    response_confirmation_timer_reached = "The timer has expired and no data will be collected."
+
+    @staticmethod
+    def _ask_opt_in(question: str, timeout: int):
+        """
+        Runs input with timeout and checks user input.
+        :param question: question that will be printed on the screen.
+        :param timeout: timeout to wait.
+        :return: opt-in dialog result.
+        """
+        colored_print(question)
+        answer = input_with_timeout(prompt='>>', timeout=timeout)
+        answer = answer.lower()
```

Comment:
I think we should also clean `answer` from white-spaces from the left and right side: `answer.lower().strip()`

Reply:
Done.